### ISSUE TITLE
Arch Linux packaging issue

### ISSUE THREAD

**Issue by horror-proton** [MaaAssistantArknights/maa-cli#158]
- Moving `share/fish/vendor_completions.d/maa.fish → completions/maa.fish` breaks install script

  https://github.com/archlinux/aur/blob/1aa219371d4a256ba384126a5b327764418ef58d/PKGBUILD#L42-L44

  https://github.com/horror-proton/PKGBUILDs/commit/0eb26b648a501eb811328ddd123a3507315e441e

----

- https://github.com/MaaAssistantArknights/maa-cli/blob/ae8f1517eaec56b5d5d3a7f2f64f9f591f2256a5/maa-cli/src/dirs.rs#L519-L525

  This test item should be removed, `MAA_EXTRA_SHARE_NAME` is not managed by maa-cli itself, it 
  would fail for requiring write permission to `/usr/share`

  ```
  ---- dirs::tests::get_dir::data_relative stdout ----
  thread 'dirs::tests::get_dir::data_relative' panicked at maa-cli/src/dirs.rs:522:53:
  called `Result::unwrap()` on an `Err` value: Os { code: 13, kind: PermissionDenied, message: 
  "Permission denied" }
  ```

**Comment by wangl-cc** [comment]
> * Moving `share/fish/vendor_completions.d/maa.fish → completions/maa.fish` breaks install script
>   https://github.com/archlinux/aur/blob/1aa219371d4a256ba384126a5b327764418ef58d/PKGBUILD#L42-L44
>   [horror-proton/PKGBUILDs@0eb26b6](https://github.com/horror-proton/PKGBUILDs/commit/0eb26b648a501eb811328ddd123a3507315e441e)

I've noticed that the file moving might affect the AUR package, so I marked it as breaking. I intend to notify you when the stable release is ready. The decision to move the file was driven by the inconvenience of the long path during development, and to ensure consistency with other related files. Furthermore, I've made changes to the build feature, which also impacts the build script. So I think it a good opportunity to change it.

> * https://github.com/MaaAssistantArknights/maa-cli/blob/ae8f1517eaec56b5d5d3a7f2f64f9f591f2256a5/maa-cli/src/dirs.rs#L519-L525
>   
>   This test item should be removed, `MAA_EXTRA_SHARE_NAME` is not managed by maa-cli itself, it
>   would fail for requiring write permission to `/usr/share`
>   ```
>   ---- dirs::tests::get_dir::data_relative stdout ----
>   thread 'dirs::tests::get_dir::data_relative' panicked at maa-cli/src/dirs.rs:522:53:
>   called `Result::unwrap()` on an `Err` value: Os { code: 13, kind: PermissionDenied, message: 
>   "Permission denied" }
>   ```

It's strange since this test should be in a temporary test directory.

https://github.com/MaaAssistantArknights/maa-cli/blob/ae8f1517eaec56b5d5d3a7f2f64f9f591f2256a5/maa-cli/src/dirs.rs#L487-L488

**Comment by horror-proton** [comment]
Oh wait, I made a mistake in PKGBUILD `MAA_EXTRA_SHARE_NAME=maa-assistant-arknights` works, while `MAA_EXTRA_SHARE_NAME=/usr/share/maa-assistant-arknights` does not

https://doc.rust-lang.org/std/path/struct.Path.html#examples-19

**Comment by wangl-cc** [comment]
It's interesting that this behavior of `join` makes it possible to specify any path as a share path.

**Comment by wangl-cc** [comment]
I think it's okay, but considering the design, `MAA_EXTRA_SHARE_NAME` should ideally be a name rather than a path. It might be worth restricting it to a path without separators?

**Comment by wangl-cc** [comment]
I released the `v0.4.0` recently. AUR build script need to be update manually.

**Comment by wangl-cc** [comment]
> > I released the `v0.4.0` recently. AUR build script need to be update manually. CC: @Cryolitia
> 
> AUR has not been maintained by me, I maintain only NUR.

This is just a mention to you for the new release, XD.

**Comment by Cryolitia** [comment]
> > > I released the `v0.4.0` recently. AUR build script need to be update manually. CC: @Cryolitia
> > 
> > AUR has not been maintained by me, I maintain only NUR.
> 
> This is just a mention to you for the new release, XD.

That's OK. ^_^
